### PR TITLE
Test-DbaLinkedServerConnection

### PR DESCRIPTION
**Pull request by ConstantineK**
Helping with a PR, should just add Test-DbaLinkedServerConnection to development

### CODE REVIEW

File: functions/Test-DbaLinkedServerConnection.ps1 [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+Test each linked server on the instance
+
+.PARAMETER SqlServer
+The SQL Server that you're connecting to.
+
+.PARAMETER SqlCredential
+Credential object used to connect to the SQL Server as a different user
+
+.NOTES
+Author: Thomas LaRock ( https://thomaslarock.com )
```
Comment (ctrlbold) [inline functions/Test-DbaLinkedServerConnection.ps1]: Lookin good!

File: functions/Test-DbaLinkedServerConnection.ps1 [hunk 2]
```diff
@@ -0,0 +1,76 @@
+function Test-DbaLinkedServerConnection
+{
+<#
+.SYNOPSIS
+Test all linked servers from the sql servers passed
+
+.DESCRIPTION
+Test each linked server on the instance
+
+.PARAMETER SqlServer
```
Comment (ctrlbold) [inline functions/Test-DbaLinkedServerConnection.ps1]: Super useful command!  I'll change this to SqlInstance for you -- it's really the only thing failing the Pester tests that ensure Get-Help describes all params.

File: functions/Test-DbaLinkedServerConnection.ps1 [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+			
+			foreach ($ls in $server.LinkedServers)
+			{ 
+      				if ($LSname -ne $null)
+      				{
+       					try
+       					{
+        					$LSname.testconnection()                     
+        					$Connectivity = $true
+        					"$(Get-Date) $ServerName $LSname connection success." | Out-File $OutputFile -Append
```
Comment (ctrlbold) [inline functions/Test-DbaLinkedServerConnection.ps1]: We do prefer for commands within the toolset to output to objects , however -- this allows end users to Out-File or Export-Csv or do whatever they like with the end result. 

To save time, I'll update this portion for you then get another team member to check my code.

File: functions/Test-DbaLinkedServerConnection.ps1 [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+			}
+			catch
+			{
+				Write-Warning "Failed to connect to: $Instance"
+				continue
+			}
+			
+			foreach ($ls in $server.LinkedServers)
+			{ 
+      				if ($LSname -ne $null)
```
Comment (ctrlbold) [inline functions/Test-DbaLinkedServerConnection.ps1]: Is there a time when $ls would not be null and thus $ls.TestConnection() would error out by default? If so, please let me know as I'm going to remove this check.

**Comment by ctrlbold** [comment]
Updated the command to output full, detailed object and updated help

image

Please let me know what you think, @SQLRockstar. Also @ConstantineK - can I get a code review? All examples passed in the lab.

**Comment by ConstantineK** [comment]
Cant officially review as I submitted the PR, but it looks good!

**Comment by ctrlbold** [comment]
(I uh, I never figured out how to consistently get just the info we need from exceptions so i left it raw) /cc @FriedrichWeinmann

**Comment by ctrlbold** [comment]
Thank you, @ConstantineK, merging :D should be in master within a few days and will be included in the next formal release.

Thanks again, @SQLRockstar! We'll update the website with the Help/Screenshots once it's formally released. I think the next release will be showmewhatyougot.

**Comment by SQLRockstar** [comment]
Awesome! Thanks for the help, and for letting me contribute. I’m looking forward to helping push this project forward!

 

 

 

Thomas LaRock
 <mailto:SQLRockstar@thomaslarock.com> SQLRockstar@thomaslarock.com
 <http://thomaslarock.com/backstage/> http://thomaslarock.com/backstage/

 <http://www.linkedin.com/in/sqlrockstar> http://www.linkedin.com/in/sqlrockstar

**Pull request merged**